src/HttpClientInterception/HttpClientInterceptorOptions.cs

```diff
@@ -405,7 +405,7 @@ public virtual HttpClient CreateHttpClient(HttpMessageHandler? innerHandler = nu
     /// </returns>
     private static string BuildKey(HttpInterceptionResponse interceptor)
     {
-        if (interceptor.UserMatcher is not null || interceptor.ContentMatcher is not null)
+        if (interceptor.UserMatcher is not null || interceptor.ContentMatcher is not null || interceptor.ConditionalMatcher is not null)
```

Comment:
I don't think we need to add another delegate here. It's still a matcher provided by the user so we can just reuse it.